drivers/include/at86rf2xx.h

```diff
@@ -207,6 +207,21 @@ extern "C" {
 
 /** @} */
 
+/**
+ * @brief Basic-extented mode TRX state mapping macros
+ */
+#if IS_ACTIVE(AT86RF2XX_BASIC_MODE)
```

Comment:
Since this is a compile time configuration, should this also change the default options that are setup in `at86rf2xx_reset`, e.f: disable auto `ACK` and `CSMA-CA`?

```
    /* set default options */
    if IS_ACTIVE(AT86RF2XX_BASIC_MODE) {
        at86rf2xx_set_option(dev, AT86RF2XX_OPT_AUTOACK, false);
        at86rf2xx_set_option(dev, AT86RF2XX_OPT_CSMA, false);

        static const netopt_enable_t enable = NETOPT_DISABLE;
        netdev_ieee802154_set(&dev->netdev, NETOPT_ACK_REQ,
                            &enable, sizeof(enable));
    }
    else {
        at86rf2xx_set_option(dev, AT86RF2XX_OPT_AUTOACK, true);
        at86rf2xx_set_option(dev, AT86RF2XX_OPT_CSMA, true);

        static const netopt_enable_t enable = NETOPT_ENABLE;
        netdev_ieee802154_set(&dev->netdev, NETOPT_ACK_REQ,
                            &enable, sizeof(enable));
    }
```

Reply:
The thing is, these configurations have no effect in basic mode.
What it's true, is the fact we should update some NETOPTs (the NETOPT_ACK_REQ doesn't make any sense in this case, so it should'nt be defined)

Reply:
> The thing is, these configurations have no effect in basic mode.
> What it's true, is the fact we should update some NETOPTs (the NETOPT_ACK_REQ doesn't make any sense in this case, so it should'nt be defined)

Yes, but if you get the `NETOPT` value it would seem like it was set right? That would be weird as well, maybe those options should just not be defined as you say, but that also means changing initialization code, so it doesn't try setting them right?

Reply:
> Yes, but if you get the NETOPT value it would seem like it was set right? That would be weird as well, maybe those options should just not be defined as you say, but that also means changing initialization code, so it doesn't try setting them right?

Sure, I can adapt that

Reply:
done!

Setting those registers in basic mode has no effect. So, I just removed those calls in Basic Mode.

Note that the "NETOPT_ACK_REQ" is a MAC thing hardcoded in the driver. It doesn't touch any register, because it's a bit used by `gnrc_netif_ieee802154`.  Strictly speaking it doesn't belong there, but for now I'm assuming it turns off while in Basic Mode. This is because the radio doesn't know how to process ACKs in Basic Mode, and we still don't have a SubMAC (#13376 , WIP) that knows what to do with that